### ISSUE TITLE
rearrange icons

### ISSUE THREAD

**Issue by nepravelniy** [will-stone/browserosaurus#297]
<img width="472" alt="Screen Shot 2020-09-11 at 18 51 06" src="https://user-images.githubusercontent.com/70824752/92946918-f81de280-f45f-11ea-8779-f103e15006ba.png">

would be great if we could rearrange browser icons
for example by drag and dropping them in the menu here:
<img width="312" alt="Screen Shot 2020-09-11 at 18 53 26" src="https://user-images.githubusercontent.com/70824752/92947040-28fe1780-f460-11ea-8354-97cddbefcbee.png">

**Comment by will-stone** [comment]
Thanks for the suggestion. I'm presuming you've numbered them like that to show that you want to move them...? Because if not, you could just renumber them 😆 I've been asked this before and my usual stance is: I just don't see the benefit gained outweighs the extra code to maintain. I'm not ruling it out completely but it's not a straightforward request because it involves keeping the order in the store, keeping it in sync with whether you install or remove browsers, not to mention (by mentioning it) the code required for drag-and-drop, testing etc.

So in conclusion, convince me of the benefits and it _may_ get added to the backlog 😃

**Comment by nepravelniy** [comment]
> I'm presuming you've numbered them like that to show that you want to move them...? Because if not, you could just renumber them 😆

Well i have sorted them by how frequent i use them - most of the time i use firefox, sometimes i use safari and almost never i touch chrome and tor (i have them just in case) so it is really convenient for me to number them the same way i have them in my head. problem starts that i can have them visually the same way and my brainlags sometimes double checking if i am pressing the right button 😄



> So in conclusion, convince me of the benefits and it may get added to the backlog 😃

I got ya. Totally valid point regarding maintenance and testing. I really love the looks and feels of the app so I wont really push you to change anything about it. However if you can make it write sequence to a small config file anywhere on the system so people can modify it, that would be awesome. 

Thinking further it can also be handy for people rocking with multiply profiles of the same browser. I do believe that such little apps like browserosaurus are super handy for developers and testers and really common stuff for that guys is to have different variations of plugins and settings which sometimes stored under different profiles.
simple config edit could also allow them to have 2 (or even more) delectable profiles right from browserosaurus as they could specify path with comments (https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Multiple_profiles). It also can have more wide usage for home people to use separate work/home profiles.

**Comment by will-stone** [comment]
> problem starts that i can have them visually the same way and my brainlags sometimes double checking if i am pressing the right button

This is why I use letters. I know whacking <kbd>space</kbd> will always get me Firefox, <kbd>e</kbd> for Edge, and <kbd>s</kbd> for Safari.

> However if you can make it write sequence to a small config file anywhere on the system so people can modify it, that would be awesome.

This is a solution, not a reason 😝 There's already some open issues for Chrome Profiles and private mode etc., so I'll say let's keep the chat about those there. I get the strong feeling that the majority of B users would want it to find and display profiles and containers automatically, but a manual config could indeed be an answer. Please have a peruse of the issues to see what's been discussed.

Okay, here's a simple solution: the browsers get sorted by the hotkey and not alphabetically. That way you can number them and know they will appear in numerical order. Any browsers without a hotkey will be sorted alphabetically at the end. The favourite will always be first. The only downside to this (and it's not a biggie) is that users who have gotten used to the current order may see them change if they are using different hotkeys other than the browsers' first letters. I think this is probably acceptable. Right... it goes on the backlog 🥳

**Comment by badcat** [comment]
> This is why I use letters. I know whacking space will always get me Firefox, **e** for Edge, and **s** for Safari.

Hmm..  I'm not sure I understand how the browsers are getting sorted by hotkey... it doesn't seem like mine are doing this.
Also - Sometimes I have different browsers that start with the same letter.

![b](https://user-images.githubusercontent.com/26096/93098200-8ff41a00-f674-11ea-8b37-8afd0d3237e7.png)

thanks for Browserosaurus!

**Comment by will-stone** [comment]
> Hmm.. I'm not sure I understand how the browsers are getting sorted by hotkey

@badcat They don't. They are currently sorted by name. They _will_ get sorted by hotkey; I was outlining the changes I will make.

**Comment by badcat** [comment]
Ahh.. hah. :) OK - sometimes English is hard for me.  thank you.

**Comment by will-stone** [comment]
No worries. Thanks for using B. And it's interesting to see someone else who uses numbers as their hotkeys: adds weight to this feature request 😄
